src/clusto/services/http.py

```diff
@@ -205,6 +205,41 @@ def show(self, request, prefetch_attrs=None):
 
         return dumps(request, result)
 
+    def rename(self, request):
+        '''
+        Renames an object and returns the new object.
+
+        Example: /server/exampleserver/rename?object=/server/newname
+        '''
+        device_type, device_name = request.params['object'].strip('/').split('/')
+        if self.obj.type != device_type:
+            return Response(status=400, body='400 Renamed object must match type of original object\n')
+        if self.obj.name == device_name:
+            # The renamed object is the same as the current object, just show the object
+            return self.show(request)
+        try:
+            obj = clusto.get_by_name(device_name)
+            return Response(status=400, body='400 An entity with the new name "%s" already exists: %s\n' % (device_name, repr(obj)))            
+        except LookupError, e:
+            pass
+        clusto.rename(self.obj.name, device_name)
```

Comment:
The reason the rename is not in the try catch block is that it is expected to throw the exception... if the exception doesn't get thrown on the lookup, it'll return with a 400 status code. I could put the rename in the exception handler block, but that doesn't feel any less ugly. I copied this pattern from the rename command implementation in the command line client.